cpp/cmake_modules/Findlibrados.cmake

```diff
@@ -0,0 +1,31 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+find_path(LIBRADOS_INCLUDE_DIR rados/librados.hpp)
+
+find_library(LIBRADOS_LIBRARY NAMES rados)
+
+mark_as_advanced(LIBRADOS_LIBRARY LIBRADOS_INCLUDE_DIR)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(librados DEFAULT_MSG LIBRADOS_LIBRARY
+                                  LIBRADOS_INCLUDE_DIR)
+
+if(LIBRADOS_FOUND)
+  set(LIBRADOS_INCLUDE_DIRS ${LIBRADOS_INCLUDE_DIR})
+  set(LIBRADOS_LIBRARIES ${LIBRADOS_LIBRARY})
```

Comment:
Could you create an import library that has these information?

```cmake
  add_library(librados::rados UNKNOWN IMPORTED)
  set_target_property(librados::rados ...)
```

See also other `Find*.cmake` in `cpp/cmake_modules/`.